include/helper_functions.php

```diff
@@ -1,15 +1,169 @@
+    
+    $remainingCredits = [];  
+    calcAllRemainingCredits($allStudents, $allCredits, $allClasses);
+ 
+    $classesBookedAmount = updateStudentInfo(getFutureClassesAmount(), sizeof($allStudents), 'Classes_Pending', '0');        //Get total classes pending per student and fills out those with no classes pending
+    $nextClasses= removeDuplicateMultiDArray(getFutureClasses(), 'Student_Id');                                              //gets the classes pending to be taught and remove any duplicate                          
+                                                                                                                             //this will keep only the date of the nearest class
+    $nextClasses = updateStudentInfo($nextClasses, sizeof($allStudents), 'Date', formatDate(currentDate(), 'Y-m-d H:i:s'));  //Fill out with current date those with not pending classes 
+    
+    $daysToNextCLass = calculateDaysToNextClass($nextClasses);  
+    
+}
+
+// --------------------------------------------------------------------------
+
+function updateStudentInfo($studentArray, $studentNumber, $value, $data){ 
+    $allInfo = [];
+    $temp = [];
+
+    $studentsArray = array_column($studentArray, 'Student_Id');                                                     //Get an array of ids of students with credits       
+    $studentArray = array_reverse($studentArray);                                                                   //Reverse it because I will use as stack below
+    
+    for($i = 0; $i < $studentNumber; $i++){
+
+        if(!in_array($i + 1, $studentsArray)){                                                                     //if the index is not in the array of ids
+            $array = array('Student_Id'=> $i + 1, $value=> $data);                                                 //set the id to 0 credit
+            array_push($allInfo, $array);                                                                          //push it to the array
+                    
+        }
+
+        else{                                                                                                      //if the index is in the array of ids
+            $temp = array_pop($studentArray);                                                                      //pop the last element (since it was reversed it is the first one in the original array)
+            array_push($allInfo, $temp);                                                                           //push it to the array
+        }
+    }
+
+    return $allInfo;
+}
+
+// --------------------------------------------------------------------------
+
+function removeDuplicateMultiDArray($array, $key) {                                                                 //function to remove duplicates from multidimentional arrays using a kew
```

Comment:
I commented in the code: I'm removing the duplicated studentId (the whole element, not only the id) because the query above returns all the future classes of each student. So there could be a student with several future classes (which means if a student has 3 future classes, there will be 3 array elements with the same studentId and just different class information). So this function removes the "duplicates" (perhaps I should name it differently). o consider any element with the same studentId to be a duplicate. In the end, it leaves only one row per student, which will be the next class, since the query is ordered by date.

Reply:
Ok, I think that last part ("since the query is ordered by date") is the part that wasn't particularly obvious to me here. I think that works, but it also seems a bit fragile. Something could change in the future to change how the array is ordered, and this functionality would break unexpectedly.

I don't think it's essential that you change it, but here are some ways you could make this more resilient:

1. Make this a more specialized function. Maybe call it `getNextClassForEachStudent`. Loop through the array and compare the date of the lesson against what's already in `$temp_array` to figure out if it should be replaced.
2. Do the same thing mentioned in #1, but make it more generic. The function name/parameters might look like `removeDuplicates($array, $keyToGroupBy, $tiebreakerFieldName, $tiebreakerCriteria="min")`. So $tiebreakerFieldName would be "dateOfClass" or whatever that field is called, and $tiebreakerCriteria would be "min" (to get the class with the lowest date for each student.

With either of those approaches, you get the same outcome, but you aren't relying on the data already being sorted a specific way.

Reply:
Ok! Just to make sure I'm clear here, if I do the date calculation in php, so, do you think I should get rid of "OrderBy date" in the query?